DataStructures/ArrayBasedCircularQueue/CircularQueue.cs

```diff
@@ -0,0 +1,153 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using DataStructures.ArrayBasedCircularQueue.Abstractions;
+using DataStructures.ArrayBasedCircularQueue.Enumerators;
+
+namespace DataStructures.ArrayBasedCircularQueue
+{
+    /// <summary>
+    /// Circular Queue based on array implementation. FIFO style. It is also called ‘Ring Buffer’.
+    /// </summary>
+    /// <remarks>
+    /// <para>Usages:</para>
+    /// <para>Memory Management.</para>
+    /// <para>Traffic system.</para>
+    /// <para>CPU Scheduling.</para>
+    /// </remarks>
+    /// <typeparam name="T">Generic Type.</typeparam>
+    public class CircularQueue<T> : IQueueOperations<T>
+    {
+        private readonly int size = 10;
+        private int front = -1;
+        private int rear = -1;
```

Comment:
Codacy found an issue: [Remove the member initializer, all constructors set an initial value for the member.](https://app.codacy.com/manual/siriak/C-Sharp/pullRequest?prid=6301193)